screens/Settings/BackupToggle.tsx

```diff
@@ -0,0 +1,68 @@
+import React, {useState} from 'react';
+import {Switch} from 'react-native-elements';
+import {Text} from '../../components/ui';
+import {Modal} from '../../components/ui/Modal';
+import {Theme} from '../../components/ui/styleUtils';
+import {useBackupScreen} from './BackupController';
+import {Platform} from 'react-native';
+import {MessageOverlay} from '../../components/MessageOverlay';
+
+export const BackupToggle: React.FC<BackupToggleProps> = props => {
+  const [dataBackup, setDataBackup] = useState(false);
+
+  const controller = useBackupScreen(props);
+
+  // TODO : Check if the setup is already done
+  const toggleSwitch = () => {
+    setDataBackup(!dataBackup);
```

Comment:
If this option triggers a backup, then writes(Writes & Updates) to MMKV & the filestorage should be frozen from the application, to ensure inconsistent changes from making mmkv corrupt, i.e. the store.ts state machine should go in a non `ready` state.